CustomSecuritySample/README.md

```diff
@@ -0,0 +1,216 @@
+
+## What Changed?
+
+A new interface is introduced that can be implemented which provides an IRSRequestContext providing the more common properties used by extensions to make decisions related to authentication. In previous version ReportManager was the front-end and could be configured with its own custom login page, in SSRS2016 only one page hosted by reportserver is supported and should authenticate to both applications.
+
+In previous versions extensions, could rely on a common assumption that ASP.NET objects would be readily available, since the new portal does not run in asp.net the extension might hit issues with objects being NULL. 
+The most generic example is accessing HttpContext.Current to read request information such as headers and cookies. In order to allow extensions to make the same decisions we introduced a new method in the extension that provides request information and is called when authenticating from the portal. 
+
+Extensions should implement the IAuthenticationExtension2 interface to leverage this. The extensions will need to implement both versions of GetUserInfo method, as is called by the reportserver context and other used in webhost process. The sample below shows one of the simple implementations for the portal where the identity resolved by the reportserver is the one used.
+  
+```csharp
+  public void GetUserInfo(IRSRequestContext requestContext, out IIdentity userIdentity, out IntPtr userId)
+       {
+           userIdentity = null;
+           if (requestContext.User != null)
+           {
+               userIdentity = requestContext.User;
+           }
+	// initialize a pointer to the current user id to zero
+           userId = IntPtr.Zero;
+       }
+```
+
+# Implementation 
+
+## Step 1: Creating the UserAccounts Database
+
+The sample includes a database script, Createuserstore.sql, that enables you to set up a user store for the Forms sample in a SQL Server database.
+Script is in the CustomSecuritySample\Setup folder.
+-	To create the UserAccounts database
+-	Open SQL Server Management Studio, and then connect to your local instance of SQL Server. 
+-	Locate the Createuserstore.sql SQL script file. The script file is contained within the sample project files. 
+-	Run the query to create the UserAccounts database. 
+-	Exit SQL Server Management Studio. 
+
+
+## Step 2: Building the Sample
+
+You must first compile and install the extension. The procedure assumes that you have installed Reporting Services to the default location: C:\Program Files\Microsoft SQL Server\MSRS13.MSSQLSERVER\Reporting Services. This location will be referred to throughout the remainder of this topic as <install_>.
+
+If you have not already created a strong name key file, generate the key file using the following instructions.
+
+To generate a strong name key file
+-	Open a Microsoft Visual Studio prompt and point to .Net Framework 4.0.
```

Comment:
I am not sure what you mean by point to .NET Framework 4.0. Maybe provide some instructions to check which version is loaded by default?